hack/generator/pkg/astmodel/resource_type.go

```diff
@@ -131,6 +131,9 @@ var _ PropertyContainer = &ResourceType{}
 // Ensure ResourceType implements the FunctionContainer interface correctly
 var _ FunctionContainer = &ResourceType{}
 
+// Ensure ObjectType implements the TestCaseDefiner interface correctly
+var _ TestCaseDefiner = &ResourceType{}
```

Comment:
I found test cases weren't being generated for resources because Resource didn't implement the interface properly. Will break this out into a separate PR as it makes sense to do so.